mpcs/src/basefold/commit_phase.rs

```diff
@@ -336,11 +337,10 @@ fn basefold_one_round_by_interpolation_weights<E: ExtensionField, Spec: Basefold
 ) -> Vec<E> {
     values
         .par_chunks_exact(2)
-        .enumerate()
-        .map(|(i, ys)| {
-            let (x0, x1, w) =
-                <Spec::EncodingScheme as EncodingScheme<E>>::prover_folding_coeffs(pp, level, i);
-            interpolate2_weights([(x0, ys[0]), (x1, ys[1])], w, challenge)
+        .zip(<Spec::EncodingScheme as EncodingScheme<E>>::prover_folding_coeffs_level(pp, level))
+        .map(|(ys, coeff)| {
+            let (lo, hi) = ((ys[0] + ys[1]).halve(), (ys[0] - ys[1]) * *coeff);
+            lo + (hi - lo) * challenge
```

Comment:
This is directly fold on evaluation (lagrange domain). 
TODO need to add document for explain this formula